### ISSUE TITLE
PermissionSelectionPartial - mismatched handler check

### ISSUE THREAD

**Issue by mmnosek** [Starcounter/Starcounter.Authorization#36]
Sometimes wrong permissions are checked while using `PermissionSelectionPartial`. 

## Steps to reproduce:
1. Create ViewModel with permission check and assign permission partial

_ViewModel.json.cs:_
```C#
[RequirePermission(typeof(ViewSth))]
partial class ViewModel {
    public ovverride void OnData() {
        base.OnData();
        PermissionPartial = PermissionSelectionPartial.Create(new EditSth(Data));
    }
}
```

_ViewModel.json:_ (Explicitly specified `PermissionPartial` type)
```json
{
    "Html": "/somepath",
    "PermissionPartial": { "$": { "Reuse": "Starcounter.Authorization.Partial.PermissionSelectionPartial" } },
}
```

2. Create other page using permission partial - almost the same as above, but without neither `RequirePermission` attribute nor `Reuse` part in JSON (replace that line by `"PermissionPartial": { }`)

3. Run app and go to the first page (with `Reuse` in JSON)
4. Go to the second page (without `Reuse` in JSON)
5. Try to trigger handler (add/remove group)
6. A unhandled exception will be thrown: `System.NullReferenceException: Object reference not set o an instance of an object.`

Basically, a system is trying to check if the user has `ViewSth` permission (it should not on this page). It's because the previously created check for `PermissionSelectionPartial` type.

## Proposed solution:
* Introduce attribute described in #35 and use it inside partial handlers
* Allow specifying which permission should be checked in the constructor (as an optional parameter). Described in: #28 

cc @joozek78

**Comment by joozek78** [comment]
PermissionSelectionPartial has been removed
